### DISCUSSION TITLE
[Idea] Related to https://github.com/PrestaShop/PrestaShop/issues/9641. Ps_emailalerts sends email for all combinations even if only 1 combination has changed the stock.

### DISCUSSION THREAD

**Discussion by cespues** [PrestaShop/PrestaShop#33333] [Ideas]
### The problem

In ps_emailalerts you can set a threshold for sending out of stock email notification

If I have 10 combinations with this threshold stock and only 1 of this combinations change the stock (1 product less), the module is sending 10 emails, even if only 1 combination has changed the stock. 

### The solution

There is some place where the product is checked when stock is updated and all combinations from this product are stored. After that the module is sending an email for each combination where the stock matches the "out_of_stock" threshold or less. 

The solution is to send one email only for the combination that have changed the stock, so there will be only 1 email in total, and not one email for each combination even if it has not changed the stock.  

I mean the solution is to check only for the id_product_attribute that has changed the stock, and not for all id_product_attribute's that belongs to an id_product. 

### Do you plan to work on this subject?

- [x] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [x] I'm willing to provide any wireframes or design assets required for this feature.
- [x] I'm willing to submit a Pull Request that implements this feature.
- [x] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by MatShir** [comment]
Great idea ! Do you feel like contributing the feature on the module ?

**Comment by cespues** [comment]
Yes!! I feel like. But it must wait until september and go on. I think
there is no rush for this, is it?

Kind regards.

**Comment by Hlavtox** [comment]
@MatShir This feels like a bug, why should it send 10 emails if only 1 combination was reduced?

We should sort it out and check it, I think that the same functionality is provided by the core itself, and maybe it works better…

**Reply by MatShir** [reply]
True, we should probably deport all into the module

**Comment by cespues** [comment]
Yes, I think this can be a bug from PS 1.7 because the module has almost
same code than MailAlerts from PS 1.6, but PS 1.7 is processing the stock
update in a different way. Now it is cheching the product_attribute_id = 0
(original product without combinations), so it works fine if this product
have 0 combinations, but when it has, is checking threshold for all
combinations, and sending email when there is a match, don't mind if the
combination has been updated.

Another functionality I added to my module (in PS 1.6) is to send only
email when threshold is == to new qty, and not when threshold is <= to new
qty, because this can mean lots of emails again, for example, if threshold
is 10, you will get an email for 10,9,8,7.....1,0. Little bit crazy. The
solution is to have only 1 or 2 thresholds (example: 10 and 0, low stock
and out of stock). So you'll get only 2 emails for
product/product_combination. With this functionality this module is really
useful and not annoying.

I can help to improve this module in PS 1.7, but it must be from 4th
september. Let me know if you want me to help in this issue (bug and/or
feature).

Kind regards.

**Reply by MatShir** [reply]
If I make strict equal, then the stock goes through from 5 quantity to 2 quantity in one order with a threshold of 4 quantity on email alert. Are you going to be able to catch it ?

**Comment by MatShir** [comment]
I have marked it as a bug so can keep track on it → https://github.com/PrestaShop/PrestaShop/issues/33474
